varcode/variant.py

```diff
@@ -46,3 +114,240 @@ def short_description(self):
                 chrom, pos + len(alt), pos + len(ref), ref[len(alt):])
         else:
             return "chr%s g.%d %s>%s" % (chrom, pos, ref, alt)
+
+    def transcripts(self):
+        return self.ensembl.transcripts_at_locus(
+            self.contig, self.pos, self.end)
+
+    def coding_transcripts(self):
+        """
+        Protein coding transcripts
+        """
+        return [
+            transcript for transcript in self.transcripts()
+            if is_coding_biotype(transcript.biotype)
+        ]
+
+    def genes(self):
+        return self.ensembl.genes_at_locus(
+            self.contig, self.pos, self.end)
+
+    def coding_genes(self):
+        """
+        Protein coding transcripts
+        """
+        return [
+            gene for gene in self.genes()
+            if is_coding_biotype(gene.biotype)
+        ]
+
+    def annotate(self, raise_on_error=True):
```

Comment:
What do you think of renaming `Annotation` (which is vague) to
`TranscriptEffectCollection` and changing the accessor method to
`variant.transcript_effects()`?

(we can add memoization if/when we find ourselves calling these methods
repeatedly and caring about the peformance)

On Fri, Feb 20, 2015 at 11:44 AM, timodonnell <notifications@github.com>
wrote:

> In varcode/variant.py
> <https://github.com/hammerlab/varcode/pull/20#discussion_r25082877>:
>
> > +        ]
> > +
> > +    def genes(self):
> > +        return self.ensembl.genes_at_locus(
> > +            self.contig, self.pos, self.end)
> > +
> > +    def coding_genes(self):
> > +        """
> > +        Protein coding transcripts
> > +        """
> > +        return [
> > +            gene for gene in self.genes()
> > +            if is_coding_biotype(gene.biotype)
> > +        ]
> > +
> > +    def annotate(self, raise_on_error=True):
>
> to me, annotate() sounds like a mutator. I'd suggest renaming this
> annotation() and memoizing it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hammerlab/varcode/pull/20/files#r25082877>.
>